backend/app/node_handler.go

```diff
@@ -147,29 +138,25 @@ func (h *Handler) ReserveNodeHandler(c *gin.Context) {
 		return
 	}
 
-	contractID, err := h.substrateClient.CreateRentContract(identity, nodeID, nil)
+	wf, err := h.ewfEngine.NewWorkflow("reserve-node")
 	if err != nil {
 		log.Error().Err(err).Send()
 		InternalServerError(c)
 		return
 	}
 
-	err = h.db.CreateUserNode(&models.UserNodes{
-		UserID:     userID,
-		ContractID: contractID,
-		NodeID:     nodeID,
-		CreatedAt:  time.Now(),
-	})
-
-	if err != nil {
-		log.Error().Err(err).Send()
-		InternalServerError(c)
-		return
+	wf.State = map[string]interface{}{
+		"user_id":  userID,
+		"mnemonic": user.Mnemonic,
+		"node_id":  nodeID,
 	}
 
-	Success(c, http.StatusOK, "Node is rented successfully", gin.H{
-		"contract_id": contractID,
+	h.ewfEngine.RunAsync(c, wf)
+
+	Success(c, http.StatusAccepted, "Node reservation in progress. You can check its status using the workflow_id.", gin.H{
```

Comment:
I think we can remove the underscore for user message in `workflow_id`
check all please